boot/bootutil/src/loader.c

```diff
@@ -780,6 +781,7 @@ boot_erase_region(const struct flash_area *fap, uint32_t off, uint32_t sz)
     return flash_area_erase(fap, off, sz);
 }
 
+#if !defined(MCUBOOT_DIRECT_XIP) && !defined(MCUBOOT_RAM_LOAD)
```

Comment:
This one is a little less clear, since the two places it is used have different ifdef blocks around them. I think it works out OK to just always have the function, and let the linker eliminate it if it isn't used.